File: c-sharp/Controllers/VigenereController.cs
using System;
using System.Collections.Generic;
using System.Text;
using Ciphers.Models;

namespace Ciphers.Controllers
{
    public class VigenereController : BaseController
    {
        public VigenereText GetVigenereText(VigenereText vText)
        {
            List<int> shiftVals = CipherService.GetShiftVals(vText.KeyWord);
            int i = 0;

            while (i < vText.PlainText.Length)
            {
                for (int j = 0; j < shiftVals.Count; j++)
                {   
                    if (i >= vText.PlainText.Length)
                    {
                        break;
                    }
                    else
                    {
                        vText.StrBuild.Append(CipherService.GetCipherLetter(
                                                                vText.PlainText[i], 
                                                                shiftVals[j]));
                    }

                    i++;
                }
            }

            vText.CipherText = vText.StrBuild.ToString();
            return vText;
        }
    }
}

File: java/src/main/java/com/metatarsal/ciphers/controllers/VigenereController.java
package com.metatarsal.ciphers.controllers;

import com.metatarsal.ciphers.models.VigenereText;
import org.springframework.stereotype.Controller;

import java.util.ArrayList;

@Controller
public class VigenereController extends BaseController {

    public VigenereText vigenereCipher(VigenereText text) {

        StringBuilder sb = getStringBuilder();
        ArrayList<Integer> shiftVals = cipherService.getShiftVals(text.getKeyword());
        int i = 0;

        while (i < text.getPlainText().length()) {

            for (int j = 0, n = shiftVals.size(); j < n; j++) {

                if (i >= text.getPlainText().length()) {
                    break;
                } else {
                    sb.append(cipherService.getCipherLetter(text.getPlainText().charAt(i),
                                                            shiftVals.get(j)));
                }

                i++;
            }
        }

        text.setCipherText(sb.toString());
        return text;
    }
}
